enterprise/server/hostedrunner/hostedrunner.go

```diff
@@ -56,12 +56,19 @@ func (r *runnerService) lookupAPIKey(ctx context.Context) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	if u.GetGroupID() == "" {
+	group := u.GetGroupID()
+	if selectedGroup != "" {
+		for _, membership := range u.GetGroupMemberships() {
+			if membership.GroupID == selectedGroup {
+				group = selectedGroup
+			}
+		}
+	}
+	if group == "" {
```

Comment:
@bduffany do we have any auth library code that will do this for us?

Reply:
Gunna submit this so I can play around with it, but will migrate if we have a better solution for getting a group id from either an HTTP or GRPC handler.

Reply:
We don't have any library code to get the preferred group ID that works for both HTTP and GRPC requests, but I could have benefited from a function like that in https://github.com/buildbuddy-io/buildbuddy/pull/1183 so I may just add a library func as part of that one.